### PR TITLE
Add group to sensor 1st discovery + template syntax {{ $xxx }}

### PR DESCRIPTION
**Pull request by PipoCanaja**
- Groups are now discovered with the 1st sensor discovery (insert) and not only on the 2nd and more (sensor update).
- Adapt the impacted testdata accordingly.
- The group name can contain `{{ $varName }}` to build it out of the precached data thanks to `YamlDiscovery::replaceValues()`.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Looks good, but I think some of the code can be simplified.  I'll check it later.

File: includes/discovery/functions.inc.php [hunk 1]
```diff
@@ -1004,6 +1005,19 @@ function discovery_process(&$valid, $device, $sensor_type, $pre_cache)
                         }
                     }
 
+                    // process the group
+                    $group = dynamic_discovery_get_value('group', $index, $data, $pre_cache);
```
[the remark this answers is not in the document]
Comment (PipoCanaja) [reply includes/discovery/functions.inc.php]: Done. Seems to be OK -BUT- I did put "null" for the $count value as it is not in use here. Any better suggestion ?

**Comment by murrant** [comment]
@PipoCanaja I required group to start with a capital letter in the validation, heh.  We'll need to udpate the validation.

**Comment by PipoCanaja** [comment]
Yep, I also allowed `{{ ` now.

File: tests/data/omnitron-iconverter.json [hunk 2]
```diff
@@ -48,11 +48,11 @@
                     "sensor_index": "pseCurrent.2",
                     "sensor_type": "omnitron-iconverter",
                     "sensor_descr": "PSE Port Ouput Current 2",
-                    "group": null,
+                    "group": "",
                     "sensor_divisor": 1000,
                     "sensor_multiplier": 1,
-                    "sensor_current": 0.07678,
-                    "sensor_limit": 0.11517,
+                    "sensor_current": 0.076780000000000001,
```
Comment (murrant) [inline tests/data/omnitron-iconverter.json]: :/ getting these wierd numbers again.  Darn floats.

**Comment by PipoCanaja** [comment]
> @PipoCanaja I required group to start with a capital letter in the validation, heh. We'll need to udpate the validation.

I changed the regexp but it seems that it was not taken. Anything else to do ?

**Comment by murrant** [comment]
@PipoCanaja just remove the regex, any string is fine.

File: tests/data/omnitron-iconverter.json (hunk 2 above)
Comment (PipoCanaja) [reply tests/data/omnitron-iconverter.json]: should I do something about it ?

**Comment by PipoCanaja** [comment]
Rhaaa, the comma ...

**Pull request merged**